File: sorting_techniques/bubble_sort.c
#include <stdio.h>

void bubble_sort(int arr[],int size)
{
	for(int i=1; i<size ;i++)
	{
		for(int j=1;j < size-i ;j++)
		{
			if(arr[j] > arr[j+1])
			{
				int temp = arr[j];
				arr[j]=arr[j+1];
				arr[j+1]=temp;
			}
	
		}
	}	
}
void print(int arr[],int size)
{
	for(int i=0;i<size;i++)
		printf("%d ",arr[i]);
	printf("\n");
}

int main()
{
	int a[5]={1,3,2,5,4};
	print(a,5);
	bubble_sort(a,5);
	print(a,5);
}

File: python_programs/bubble_sort.py

def bubblesort(array):
    n = len(array)
    for i in range(n):

        for j in range(0,n - i - 1):

            if array[j] > array[j+1]:

                temp = array[j]
                array[j] = array[j+1]
                array[j+1] = temp

def main():
    numbers = [2,1,3,5,4]
    print("Before sort:",numbers)
    bubblesort(numbers)
    print("After sort:",numbers)

if __name__ == "__main__":
    main()


